### PR TITLE
Clarifications to the serving overview

### PR DESCRIPTION
**Pull request by sarahmaddox**
Clarifies the serving overview:
* Adds an introductory paragraph to the page.
* Shortens and clarifies some sentences.
* Styles the table.
* Changes Xs to check marks and moves the links to aliases

Fixes https://github.com/kubeflow/website/issues/1459

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1454)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-1454--competent-brattain-de2d6d.netlify.com/docs/components/serving/overview/

**Comment by sarahmaddox** [comment]
/assign @ellis-bigelow 

@ellis-bigelow Please would you take a look at this PR - it's the promised followup on the original PR https://github.com/kubeflow/website/pull/1405

**Comment by sarahmaddox** [comment]
This PR is now ready for review. I'm applying a hold to allow for multiple reviewers:
/hold

File: content/docs/components/serving/overview.md [hunk 1]
```diff
@@ -4,62 +4,208 @@ description = "Model serving overview"
...
+requirements.
 
+## Multi-framework serving with KFServing or Seldon Core
+
+KFServing and Seldon Core are both open source systems that allow 
+multi-framework model serving. The following table compares 
+KFServing and Seldon Core. A check mark (**&check;**) indicates that the system
+(KFServing or Seldon Core) supports the feature specified in that row.
+
+<div class="table-responsive">
```
Comment (ellis-bigelow) [inline content/docs/components/serving/overview.md]: This isn't rendering correctly. Are you sure it's correct? Otherwise LGTM

Comment (sarahmaddox) [reply content/docs/components/serving/overview.md]: Thanks @ellis-bigelow. The table looks OK on my screen. What rendering problem is it causing on your side?
https://deploy-preview-1454--competent-brattain-de2d6d.netlify.com/docs/components/serving/overview/

**Pull request merged**